app/core/redis_manager.py

```
@@ -335,14 +335,25 @@ async def _perform_health_check(self):
+            
+            # Retrieve and verify data integrity
             result = await self._redis.get(test_key)
+            
+            # Clean up test data
             await self._redis.delete(test_key)
             
-            if result != b"test_value":
-                raise ValueError("Health check data integrity failure")
+            # Verify data integrity with proper byte comparison
+            if result is None:
+                raise ValueError("Health check data retrieval failure - key not found")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)